internal/trace/transport.go

```diff
@@ -16,56 +16,58 @@ limitations under the License.
 package trace
 
 import (
+	"fmt"
 	"net/http"
 	"strings"
-
-	"github.com/sirupsen/logrus"
+	"sync/atomic"
 )
 
 // Transport is an http.RoundTripper that keeps track of the in-flight
 // request and add hooks to report HTTP tracing events.
 type Transport struct {
 	http.RoundTripper
+	count uint64
 }
 
 func NewTransport(base http.RoundTripper) *Transport {
-	return &Transport{base}
+	return &Transport{base, 0}
 }
 
 // RoundTrip calls base roundtrip while keeping track of the current request.
 func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	number := atomic.AddUint64(&t.count, 1)
 	ctx := req.Context()
 	e := Logger(ctx)
 
-	e.Debugf("> Request URL: %q", req.URL)
-	e.Debugf("> Request method: %q", req.Method)
-	e.Debugf("> Request headers:")
-	logHeader(req.Header, e)
+	// log the request
+	e.Debugf("\nRequest #%d\n> Request URL: %q\n> Request method: %q\n> Request headers:\n%s",
```

Comment:
Having counting number in the log message is a straightforward way to show the correlation info. Please also consider is it possible to add logging fields with type (request/response) and count (incremental number of http request connections)? Do we need to make any change in `context.go` to make sure all fields are shown in all logs?

Reply:
Looks like if the output goes to a terminal, then the log fields will not be printed. Since #851 is targeting to improve developer experience, we need to keep the type and count in the log message.

Reply:
We decided to keep the counting number in the message